test/integration/scheduler_perf/scheduler_perf.go

```diff
@@ -1041,6 +1055,13 @@ func setupTestCase(t testing.TB, tc *testCase, featureGates map[featuregate.Feat
 	timeout := 30 * time.Minute
 	tCtx = ktesting.WithTimeout(tCtx, timeout, fmt.Sprintf("timed out after the %s per-test timeout", timeout))
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.SchedulerQueueingHints) {
```

Comment:
`thresholdMetricsSelector` `isValid` method uses `defaultMetricsCollectorConfig` to verify if the correct metric is selected. It is done on validation level, so before this step.

https://github.com/kubernetes/kubernetes/blob/efd2cd53bdd1671a3532a48698f22ae1b8192cda/test/integration/scheduler_perf/scheduler_perf.go#L338-L358

Reply:
moved `isValid` so that it'll be done after the QHint metric registration.